extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleJobStore.java

```diff
@@ -251,4 +265,73 @@ private Key getKey(UUID jobId) {
 
     return builder.build();
   }
-}
\ No newline at end of file
+
+  @Override
+  public <T extends DataModel> void create(UUID jobId, T model) throws IOException {
+    Preconditions.checkNotNull(jobId);
+    Transaction transaction = datastore.newTransaction();
+    Key key = getDataKey(jobId, model.getClass());
+    Entity shouldNotExist = transaction.get(key);
+    if (shouldNotExist != null) {
+      transaction.rollback();
+      throw new IOException(
+          "Record already exists for data key: " + key.getName() + ". Record: " + shouldNotExist);
+    }
+
+    String serialized = objectMapper.writeValueAsString(model);
+    Entity entity = Entity.newBuilder(key)
+        .set(CREATED_FIELD, Timestamp.now())
+        .set(model.getClass().getName(), serialized)
+        .build();
+
+    try {
+      transaction.put(entity);
+    } catch (DatastoreException e) {
+      transaction.rollback();
+      throw new IOException(
+          "Could not create initial record for jobID: " + jobId + ". Record: " + entity, e);
```

Comment:
Can you log data type here as well? or log the key as you do below, since that should include job ID and data type